### ISSUE TITLE
Next version after 2023.05.15

### ISSUE THREAD

**Issue by moving-bits** [cgeo/cgeo#14283]
Let's collect issues for upcoming bugfix release(s) after current release 2023.05.15:

Breaking bugs (crashes etc.):
- [x] #14278
- [ ] #14279
- [ ] #14293
- [ ] #14294
- [ ] #14295
- [x] #14328

Non-breaking bugs and nice-to-haves:
- [x] #14274

Feel free to edit/update this lists as required.

**Comment by moving-bits** [comment]
I propose to create a new bugfix version from current state of `release` tomorrow.

Although not all of the issues in the list above are fixed already, we have quite a few useful fixes on `release` (see [changelog](https://github.com/cgeo/cgeo/blob/release/main/src/main/res/raw/changelog_bugfix.md)), some of them addressing the highest-ranked issues currently.

I'll start working on it Sunday evening, if no one stops me.

**Comment by fm-sys** [comment]
> I'll start working on it Sunday evening, if no one stops me.

We also have our dev meeting tomorrow, don't we?

**Comment by moving-bits** [comment]
As discussed in yesterday's meeting: Create new bugfix release _after_ having implemented the following two things:
- #14328
- extended stack trace for #14295

**Comment by moving-bits** [comment]
As discussed in our last team meeting: Will start working on a new bugfix release now:

- [ ] Check/insert correct version name in `BranchDetectionHelper.java`
- [ ] Wait for a Crowdin cycle or trigger sync manually
- [ ] Merge the related Crowdin translation pull requests for branch `release`
- [ ] Build release on CI server
- [ ] Perform smoke test with resulting ```cgeo-release.apk``` before continuing
- [ ] Upload and publish ```cgeo-release.apk``` to Google Play
- [ ] Tag the release on Github in format `market_yyyymmdd`
- [ ] Publish the release on Github including upload of ```cgeo-release.apk``` to Github
- [ ] Publish release on F-Droid
- [ ] Trigger notification to status.cgeo.org via CI server
- [ ] Post release info to Facebook 
- [ ] Post release info to Twitter

**Comment by fm-sys** [comment]
IMHO we should start a new beta cycle rather soon, as the next version of the [OAM elevate theme will require mapsforge 0.19](https://www.openandromaps.org/oam-forums/topic/darstellung-von-radschnellwegen/page/2#post-54670) which we have new at master (PR  #14414).

Do we have any larger new features in the pipeline currently? Anything broken which needs to be fixed before we can start a new beta?

**Comment by moving-bits** [comment]
I'm not aware of any larger new feature in the pipeline currently. Not sure if @eddiemuc might be working on #14375, though.

Otherwise I will start preparing a beta release tomorrow (Saturday) evening, including the Twitter PR (if no one objects).

**Comment by eddiemuc** [comment]
> Not sure if @eddiemuc might be working on #14375, though.

I don't

**Comment by moving-bits** [comment]
I'll start preparing a new beta version now:

- [ ] Empty `changelog_bugfix.md` on branch `master`
- [ ] Wait for a Crowdin cycle or trigger sync manually for branch `master`
- [ ] Merge the related Crowdin translation pull requests for branch `master`
- [ ] Delete outdated translations branch for `release`
- [ ] Fast forward `release` branch to HEAD of `master`
- [ ] Check/insert correct version name in `BranchDetectionHelper.java` on branch `release`
- [ ] Build release on CI server
- [ ] Perform smoke test with resulting ```cgeo-RC.apk``` before continuing
- [ ] Upload and publish ```cgeo-RC.apk``` to Google Play on beta channel
- [ ] Trigger notification to status.cgeo.org via CI server
- [ ] Post release info to Facebook 
- [ ] Post release info to Twitter

**Comment by moving-bits** [comment]
As discussed in our dev meeting last week I'll start preparing our next release now:

- [ ] Check/insert correct version name in `BranchDetectionHelper.java`
- [ ] Wait for a Crowdin cycle or trigger sync manually
- [ ] Merge the related Crowdin translation pull requests for branch `release`
- [ ] Build release on CI server
- [ ] Perform smoke test with resulting ```cgeo-release.apk``` before continuing
- [ ] Upload and publish ```cgeo-release.apk``` to Google Play
- [ ] Tag the release on Github in format `market_yyyymmdd`
- [ ] Publish the release on Github including upload of ```cgeo-release.apk``` to Github
- [ ] Publish release on F-Droid
- [ ] Trigger notification to status.cgeo.org via CI server
- [ ] Post release info to Facebook 
- [ ] Post release info to Twitter

**Comment by moving-bits** [comment]
Will start preparing a bugfix release now, as discussed in #14499:

- [ ] Check/insert correct version name in `BranchDetectionHelper.java`
- [ ] Wait for a Crowdin cycle or trigger sync manually
- [ ] Merge the related Crowdin translation pull requests for branch `release`
- [ ] Build release on CI server
- [ ] Perform smoke test with resulting ```cgeo-release.apk``` before continuing
- [ ] Upload and publish ```cgeo-release.apk``` to Google Play
- [ ] Tag the release on Github in format `market_yyyymmdd`
- [ ] Publish the release on Github including upload of ```cgeo-release.apk``` to Github
- [ ] Publish release on F-Droid
- [ ] Trigger notification to status.cgeo.org via CI server
- [ ] Post release info to Facebook 
- [ ] Post release info to Twitter

**Comment by moving-bits** [comment]
With PR #14580 pending I'd like to raise the question about a new release, either bugfix or feature release.

We do have some minor fixes on `release` branch which are not yet released:
- navigation bar fix
- fav ratio fix
- some icon scaling fixes
- GPX import with multiple tracks fix

Before we can release this, we will have to set `targetSDK` to 33. This is in the nightlies for quite some time now, but user base for the nightlies is relatively small, so we should get some more user feedback, IMHO.

Thus I propose to
- set `targetSDK` to 33 on `release`
- and create a new beta from `release` to get some more feedback on possible "targetSDK 33 quirks"

Other options would be:
- do not release for the time being, just continue development (including Material design) on `master`
- branch current state of `master` and create a beta version from it (for an upcoming full release), then merge PR #14580

What do you think?

**Comment by bekuno** [comment]
We should have a `release` branch to react fast on urgent changes.
On `master` we have currently too much unfinished changes.
Therefore I support variant one including a beta version.

**Comment by moving-bits** [comment]
I'll start working on a new beta from `release` branch to get a broader testing base for `targetSDK` 33 (and to be able to do releases again).

- [ ] Check/insert correct version name in `BranchDetectionHelper.java`
- [ ] Wait for a Crowdin cycle or trigger sync manually
- [ ] Merge the related Crowdin translation pull requests for branch `release`
- [ ] Build release on CI server
- [ ] Perform smoke test with resulting ```cgeo-RC.apk``` before continuing
- [ ] Upload and publish ```cgeo-RC.apk``` to Google Play on beta channel
- [ ] Trigger notification to status.cgeo.org via CI server
- [ ] Post release info to Facebook 
- [ ] Post release info to Twitter

**Comment by moving-bits** [comment]
Nothing has come up yet WRT to our beta version, neither as beta feedback nor via support, so `targetSDK` 33 handling seems to be fine. Therefore I plan to prepare a (bugfix) release from current beta version on Wednesday.

**Comment by moving-bits** [comment]
Will start preparing a new beta today, as proposed in https://github.com/cgeo/cgeo/issues/14633#issuecomment-1725897518

- [ ] Check/insert correct version name in `BranchDetectionHelper.java`
- [ ] Wait for a Crowdin cycle or trigger sync manually
- [ ] Merge the related Crowdin translation pull requests for branch `release`
- [ ] Build release on CI server
- [ ] Perform smoke test with resulting ```cgeo-RC.apk``` before continuing
- [ ] Upload and publish ```cgeo-RC.apk``` to Google Play
- [ ] Upload and publish ```cgeo-RC.apk``` to Google Play on beta channel
- [ ] Trigger notification to status.cgeo.org via CI server
- [ ] Post release info to Facebook 
- [ ] Post release info to Twitter

If nothing new comes up, a bugfix release could follow on Friday.

**Comment by alan666notb** [comment]
@moving-bits - I closed #14633 as it was fixed in the latest beta, but I realise that people really need a warning.

My partner updated to the latest beta but then said "They're still not working" - because she hadn't run the wizard to allow notifications.

Trouble is, notifications get switched off silently (Maybe by targetSDK 33?) so there's no indication that anything needs to be done.

**Comment by ztNFny** [comment]
> I vote for opening the wizard automatically (once while upgrading)

maybe for SDK >= 33 only?

**Comment by bekuno** [comment]
> I've created #14641 for this. Shall I update beta version tomorrow (Friday)?

That makes sense.

**Comment by moving-bits** [comment]
Updating beta version now...

- [ ] Check/insert correct version name in `BranchDetectionHelper.java`
- [ ] Wait for a Crowdin cycle or trigger sync manually
- [ ] Merge the related Crowdin translation pull requests for branch `release`
- [ ] Build release on CI server
- [ ] Perform smoke test with resulting ```cgeo-RC.apk``` before continuing
- [ ] Upload and publish ```cgeo-RC.apk``` to Google Play on beta channel
- [ ] Trigger notification to status.cgeo.org via CI server
- [ ] Post release info to Facebook 
- [ ] Post release info to Twitter

If nothing new comes up, a bugfix release could follow on Monday.

**Comment by moving-bits** [comment]
Nothing serious came up on support for our current beta, so I'll start preparing next bugfix release now. This will be the first release targeting SDK 33.

- [ ] Check/insert correct version name in `BranchDetectionHelper.java`
- [ ] Wait for a Crowdin cycle or trigger sync manually
- [ ] Merge the related Crowdin translation pull requests for branch `release`
- [ ] Build release on CI server
- [ ] Perform smoke test with resulting ```cgeo-release.apk``` before continuing
- [ ] Upload and publish ```cgeo-release.apk``` to Google Play
- [ ] Tag the release on Github in format `market_yyyymmdd`
- [ ] Publish the release on Github including upload of ```cgeo-release.apk``` to Github
- [ ] Publish release on F-Droid
- [ ] Trigger notification to status.cgeo.org via CI server
- [ ] Post release info to Facebook 
- [ ] Post release info to Twitter

**Comment by moving-bits** [comment]
New feature release will enter beta phase soon, so let's close this old issue here.
